backend-controller/entryController.js

```diff
@@ -12,36 +10,93 @@ class DiaryController {
     res.send('Andela BootCamp Challenge 2.....Welcome to My Diary Api Endpoints Written by Obuladike Chisom.....');
   }
 
-  static createEntry(req, res) {
+  static createEntry(req, res, next) {
     const { error } = validateEntry(req.body);
-    if (error) res.status(400).json({ error: error.details[0].message });
-    const id = allEntries.length + 1;
-    const entryAdded = new CreateEntry(id, req.body.title, req.body.category, req.body.sub_category,
-      req.body.content);
-    allEntries.push(entryAdded);
-    res.status(201).json({ success: 'success', entry: entryAdded });
-  }
-
-  static editEntry(req, res) {
-    const entryToModify = allEntries.find(e => e.id === parseInt(req.params.id, 10));
-    if (!entryToModify || entryToModify === undefined) {
-      res.status(404).json({
-        error: 'The entry you wish to modify must have been removed or have not been created',
+    if (error) {
+      res.status(400).json({ error: error.details[0].message });
+      return;
+    }
+    const check = auth.verifyUserToken(req);
+    if (check === 401) {
+      res.status(401).json({ error: 'No token provided.' });
+    } else if (check === 500) {
+      res.status(500).json({ error: 'Failed to authenticate token.' });
+    } else {
+      const query = {
+        text: 'Select * from users where id = $1 LIMIT 1',
+        values: [check.id],
+      };
+      db.query(query, (error2, response) => {
+        if (error2) return next(error2);
+        if (response.rows.length > 0) {
+          const query2 = {
+            text: `INSERT INTO entries (user_id, title, category, sub_category, content)
+             VALUES ($1,$2,$3,$4,$5)`,
+            values: [
+              check.id, req.body.title, req.body.category, req.body.sub_category,
+              req.body.content,
+            ],
+          };
+          db.query(query2, (error3, res3) => {
+            if (error3) return next(error3);
+            return res.status(201).json({ success: 'Entry Created successfully' });
+          });
+        } else {
+          return res.status(404).json({ error: 'User not found' });
+        }
       });
     }
+  }
+
+  static editEntry(req, res, next) {
     const { error } = validateEntry(req.body);
     if (error) {
       res.status(400).json({ error: error.details[0].message });
       return;
     }
-    entryToModify.title = req.body.title;
-    entryToModify.category = req.body.category;
-    entryToModify.sub_category = req.body.sub_category;
-    entryToModify.content = req.body.content;
-    res.status(200).json({ success: 'success', entry: entryToModify });
+    const check = auth.verifyUserToken(req);
```

Comment:
@obulaworld  Please take all validation to middleware, apply every other place

Reply:
okay..thanks